File: apps/backend/app/schemas/milestone.py
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, ConfigDict

# --- Admin / público -----------------------------------------
class MilestoneBase(BaseModel):
    title: str
    description: str | None = None
    points: int
    order: int = 0
    active: bool = True

class MilestoneCreate(MilestoneBase):
    pass

class MilestoneUpdate(MilestoneBase):
    pass

class MilestoneRead(MilestoneBase):
    id: UUID
    image_url: str
    created_at: datetime
    updated_at: datetime
    model_config = ConfigDict(from_attributes=True)

class PaginatedMilestone(BaseModel):
    total: int
    skip: int
    limit: int
    items: list[MilestoneRead]
    model_config = ConfigDict(from_attributes=True)

# --- Para o usuário ------------------------------------------
class UserMilestoneRead(BaseModel):
    milestone: MilestoneRead
    achieved_at: datetime
    model_config = ConfigDict(from_attributes=True)


# app/schemas/milestone.py  (ou onde definiu)

class NextMilestoneRead(BaseModel):
    milestone_id: UUID
    title: str
    points: int          # <- era threshold
    image_url: str
    user_points: int
    remaining: int

class MilestoneStatusRead(BaseModel):
    id: UUID
    title: str
    points: int          # <- era threshold
    image_url: str
    achieved: bool
    achieved_at: datetime | None

File: apps/portal-client/src/types/milestone.ts
// src/types/milestone.ts

export interface MilestoneBase {
  title: string
  description?: string | null
  points: number
  order?: number
  active?: boolean
}

export interface MilestoneCreate extends MilestoneBase {
  // image virá por multipart/form-data
}

export interface MilestoneUpdate extends MilestoneBase {
  // image opcional
}

export interface MilestoneRead extends MilestoneBase {
  id: string
  image_url: string
  created_at: string
  updated_at: string
}

export interface PaginatedMilestone {
  total: number
  skip: number
  limit: number
  items: MilestoneRead[]
}

export interface UserMilestoneRead {
  milestone: MilestoneRead
  achieved_at: string
}


export interface NextMilestoneRead {
  milestone_id: string;   // UUID
  title: string;
  points: number;      // pontos necessários
  image_url: string;
  user_points: number;    // pontos atuais do usuário
  remaining: number;      // pontos faltantes
}

/** Listagem completa de marcos com status para o usuário logado */
export interface MilestoneStatusRead {
  id: string;             // UUID
  title: string;
  points: number;
  image_url: string;
  achieved: boolean;
  achieved_at: string | null;  // ISO-8601 ou null
}